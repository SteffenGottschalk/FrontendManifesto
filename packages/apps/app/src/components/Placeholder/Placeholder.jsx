import React from 'react';
import PropTypes from 'prop-types';

const Placeholder = ({ text, color, disabeld }) => {
  return (
    <div
      style={{
        borderLeft: `solid 8px ${color}`,
        padding: '8px',
        opacity: disabeld ? 0.5 : 1
      }}>
      {text}
    </div>
  );
};

Placeholder.propTypes = {
  text: PropTypes.string,
  color: PropTypes.string,
  disabeld: PropTypes.bool
};

Placeholder.defaultProps = {
  text: '',
  color: 'green',
  disabeld: false
};

export default Placeholder;
