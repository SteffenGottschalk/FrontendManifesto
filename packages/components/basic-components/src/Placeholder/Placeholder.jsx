import React from "react";
import PropTypes from "prop-types";

const Placeholder = ({ text, color }) => {
  return (
    <div style={{ color: `${color}` }}>
      ddsfsdfsdf
      {text}
    </div>
  );
};

Placeholder.propTypes = {
  text: PropTypes.string,
  color: PropTypes.string,
};

Placeholder.defaultProps = {
  text: "",
  color: "yellow",
};

export default Placeholder;
