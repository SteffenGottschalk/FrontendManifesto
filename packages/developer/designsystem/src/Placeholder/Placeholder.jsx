import React from 'react';
import PropTypes from 'prop-types';

const Placeholder = ({ text, color }) => {
  return <div style={{ color: `${color}` }}>{text}</div>;
};

Placeholder.propTypes = {
  text: PropTypes.string,
  color: PropTypes.string,
};

Placeholder.defaultProps = {
  text: '',
  color: 'red',
};

export default Placeholder;
