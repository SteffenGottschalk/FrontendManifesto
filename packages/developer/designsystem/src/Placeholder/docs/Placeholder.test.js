import React from 'react';
import { render, screen } from '@testing-library/react';

import Placeholder from '../Placeholder';

describe('Placeholder Component', () => {
  // Smoke Test
  it('renders without crashing', () => {
    expect(() => render(<Placeholder />)).not.toThrow();
  });
  it('renders with correct Label', () => {
    render(<Placeholder text="My Label Text" />);
    expect(screen.getByText('My Label Text')).toBeInTheDocument();
  });
});
