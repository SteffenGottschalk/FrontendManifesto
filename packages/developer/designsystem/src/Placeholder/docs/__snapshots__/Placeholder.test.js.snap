// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Component renders with correct Labelan 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  My Label Text
</div>
`;
